### ISSUE TITLE
Movie stays snatched

### ISSUE THREAD

**Issue by theflu** [nosmokingbandit/Watcher3#33]
I am using sabnzbd and transmission. Watcher snatches the file just fine, but when it finished nothing happens. The status stays set to snatched and the file never gets renamed or moved. Both of my downloaders are on the same both that Watcher3 is on.

A side not when I mark a file as bad in watcher it doesn't cancel the download.

**Comment by theflu** [comment]
Error from sabnzbd post processing script.
```
Sending file%20name to Watcher as Complete.
Traceback (most recent call last):
  File "/root/NZBtoMedia/sabnzbd.py", line 79, in <module>
    request = request(url, post_data, headers={'User-Agent': 'Mozilla/5.0'})
AttributeError: Request instance has no __call__ method
```

**Comment by theflu** [comment]
Fixed sabnzbd.py error by changing lines 79, 80
From:
```
request = request(url, post_data, headers={'User-Agent': 'Mozilla/5.0'})
response = json.loads(urlopen(request, timeout=600).read().decode('utf-8'))
```
To:
```
request = urllib2.Request(url, post_data, headers={'User-Agent': 'Mozilla/5.0'})
response = json.loads(urllib2.urlopen(request, timeout=600).read().decode('utf-8'))
```

**Comment by theflu** [comment]
Another error from sabnzdb.py
```
Sending file%20name to Watcher as Complete.
Traceback (most recent call last):
  File "/root/NZBtoMedia/sabnzbd.py", line 80, in <module>
    response = json.loads(urllib2.urlopen(request, timeout=600).read().decode('utf-8'))
  File "/usr/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 500: Internal Server Error
```

**Comment by theflu** [comment]
The transmission post-processing  script seems to working fine.

**Comment by theflu** [comment]
Another error from Watcher logs for a different movie:
```
IndexError: list index out of range
    search_result = self.score.score([search_result], imported=True)[0]
  File "/opt/Watcher3/core/library.py", line 648, in searchresults
    if self.update.searchresults(data['guid'], 'Finished', movie_info=data):
  File "/opt/Watcher3/core/postprocessing.py", line 363, in complete
    response = self.complete(data)
  File "/opt/Watcher3/core/postprocessing.py", line 99, in POST
    return self.callable(*self.args, **self.kwargs)
  File "/opt/Watcher3/lib/cherrypy/_cpdispatch.py", line 60, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/opt/Watcher3/lib/cherrypy/lib/encoding.py", line 220, in __call__
    response.body = self.handler()
  File "/opt/Watcher3/lib/cherrypy/_cprequest.py", line 670, in respond
Traceback (most recent call last):
ERROR 2017-04-26 14:15:55,092 cherrypy.error.139624341090992.error: [26/Apr/2017:14:15:55] HTTP 
  CONNECTION: close
  USER-AGENT: Mozilla/5.0
  Content-Length: 307
  Remote-Addr: 127.0.0.1
  Content-Type: application/x-www-form-urlencoded
  HOST: localhost:9090
  ACCEPT-ENCODING: identity
Request Headers:
```

**Comment by theflu** [comment]
Latest update seems to have fixed the issues I was having.

**Comment by nosmokingbandit** [comment]
Sorry I never responded directly. I got behind on emails and didn't mean to ignore you.

45fe5f7c91eb8fab3a3670a020e61cc7bb2cbdfb has a fixed sab post-script that avoids the error you first posted.

As far as the other problem, I did rework a lot of post-processing methods which would have affected the issue in your logs.

If everything is cool I'll close this, if not let me know what is happening and I'll take care of it.
